### PR TITLE
cortexm.inc.mk: detect 'arm-none-eabi-gcc' from ubuntu bionic

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

arm-gcc version from ubuntu bionic repository is not supported in RIOT.
Both when building with `gnu` and `llvm`.
    
    arm-none-eabi-gcc --version
    arm-none-eabi-gcc (15:6.3.1+svn253039-1build1) 6.3.1 20170620
    
So detect this version and print an error if found.
    
The check is done on the building machine and not on the host when building in
docker.
    
The error can be disabled when building with WERROR=0.

### TODO

* [ ] Verify it does not fail on murdock "test nodes" when doing only "flash test" 
    * [ ] maybe catch if it is not installed ?
* [ ] verify mac does not break, the 'sed' command should be safe.

#### Not matching your working gcc version

Try compiling anything for `samr21-xpro` with your working setup, it should produce no error at compilation

#### Detecting bionic

Try building with ubuntu bionic GCC version (change your PATH) to see the error is detected, and also other "working" gcc versions:

```
RIOT_CI_BUILD=1 BOARD=samr21-xpro make --no-print-directory -C examples/default/ 
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:61: arm-none-eabi-gcc version not supported
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:62: arm-none-eabi-gcc (15:6.3.1+svn253039-1build1) 6.3.1 20170620
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:63: The currently recommanded version is the one installed in the riotdocker image
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:64: https://github.com/RIOT-OS/riotdocker/blob/master/Dockerfile
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:66: *** This check can be ignored by building with 'WERROR=0'.  Stop.
```

You can disable the error with `WERROR=0` and still try compiling:

```
WERROR=0 RIOT_CI_BUILD=1 BOARD=samr21-xpro make --no-print-directory -C examples/default/ 
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:61: arm-none-eabi-gcc version not supported
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:62: arm-none-eabi-gcc (15:6.3.1+svn253039-1build1) 6.3.1 20170620
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:63: The currently recommanded version is the one installed in the riotdocker image
/home/harter/work/git/RIOT/makefiles/arch/cortexm.inc.mk:64: https://github.com/RIOT-OS/riotdocker/blob/master/Dockerfile
Building application "default" for "samr21-xpro" with MCU "samd21".

   text    data     bss     dec     hex filename
  47436     148    6044   53628    d17c /home/harter/work/git/RIOT/examples/default/bin/samr21-xpro/default.elf
```

When building in docker, no check is done on your machine:

```
BUILD_IN_DOCKER=1 DOCKER="sudo docker" RIOT_CI_BUILD=1 BOARD=samr21-xpro make --no-print-directory -C examples/default/ 
Launching build container using image "riot/riotbuild:latest".
sudo docker run --rm -t -u "$(id -u)" \
    -v '/home/harter/work/git/RIOT:/data/riotbuild/riotbase' \
    -v '/home/harter/work/git/RIOT/cpu:/data/riotbuild/riotcpu' \
    -v '/home/harter/work/git/RIOT/boards:/data/riotbuild/riotboard' \
    -v '/home/harter/work/git/RIOT/makefiles:/data/riotbuild/riotmake' \
    -v '/home/harter/work/git/RIOT:/data/riotbuild/riotproject' \
    -v /etc/localtime:/etc/localtime:ro \
    -e 'RIOTBASE=/data/riotbuild/riotbase' \
    -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' \
    -e 'RIOTCPU=/data/riotbuild/riotcpu' \
    -e 'RIOTBOARD=/data/riotbuild/riotboard' \
    -e 'RIOTMAKE=/data/riotbuild/riotmake' \
    -e 'RIOTPROJECT=/data/riotbuild/riotproject' \
    -v /home/harter/.gitcache:/data/riotbuild/gitcache -e GIT_CACHE_DIR=/data/riotbuild/gitcache  \
    -e 'BOARD=samr21-xpro' -e 'RIOT_CI_BUILD=1' \
    -w '/data/riotbuild/riotproject/examples/default/' \
    'riot/riotbuild:latest' make  
Building application "default" for "samr21-xpro" with MCU "samd21".

   text    data     bss     dec     hex filename
  39796     508    6044   46348    b50c /data/riotbuild/riotproject/examples/default/bin/samr21-xpro/default.elf
```

### Issues/PRs references

Was reported here:

* https://github.com/RIOT-OS/RIOT/issues/9248
* https://github.com/RIOT-OS/RIOT/issues/9846
* https://github.com/RIOT-OS/RIOT/issues/9949
* https://github.com/RIOT-OS/riotdocker/pull/55#issuecomment-428513572

### CODE REVIEW

**Comment by cladmi** [comment]
@HendrikVE could you also try ?

**Comment by HendrikVE** [comment]
Works good for me, warning is displayed and bypass with WERROR=0 also works.

File: makefiles/arch/cortexm.inc.mk [hunk 1]
```diff
@@ -43,6 +43,32 @@ export USEMODULE += newlib
...
+# Both when using gnu and llvm toolchains
+#
+#     /usr/bin/arm-none-eabi-gcc --version | head -n 1
+#     arm-none-eabi-gcc (15:6.3.1+svn253039-1build1) 6.3.1 20170620
+#
+ARM_GCC_UNSUPPORTED += 15:6.3.1%  # ubuntu bionic, ignore 'svn' build part
+
+# Should not raise an error on the host system version when building in docker
+ifneq (1,$(BUILD_IN_DOCKER))
+ifneq (,$(filter $(ARM_GCC_UNSUPPORTED),$(ARM_GCC_VERSION)))
```
Comment (miri64) [inline makefiles/arch/cortexm.inc.mk]: Maybe use indentation here to make it more readable:

```suggestion
  ifneq (,$(filter $(ARM_GCC_UNSUPPORTED),$(ARM_GCC_VERSION)))
```

Comment (cladmi) [reply makefiles/arch/cortexm.inc.mk]: Sure, I update all lines.

**Comment by cladmi** [comment]
Updated (not sure it sent a message)

File: makefiles/arch/cortexm.inc.mk (hunk 1 above)
Comment (miri64) [reply makefiles/arch/cortexm.inc.mk]: Was addressed.

File: makefiles/arch/cortexm.inc.mk [hunk 2]
```diff
@@ -43,6 +43,32 @@ export USEMODULE += newlib
...
+ARM_GCC_UNSUPPORTED += 15:6.3.1%  # ubuntu bionic, ignore 'svn' build part
+
+# Should not raise an error on the host system version when building in docker
+ifneq (1,$(BUILD_IN_DOCKER))
+  ifneq (,$(filter $(ARM_GCC_UNSUPPORTED),$(ARM_GCC_VERSION)))
+    $(warning $(TARGET_ARCH)-gcc version not supported)
+    $(warning $(shell $(TARGET_ARCH)-gcc --version | head -n 1))
+    $(warning The currently recommanded version is the one installed in the riotdocker image)
+    $(warning https://github.com/RIOT-OS/riotdocker/blob/master/Dockerfile)
+    ifeq (1,$(WERROR))
```
Comment (miri64) [inline makefiles/arch/cortexm.inc.mk]: From the error message below I would expect it to be checked with `ifneq (0,$(WERROR))`

Comment (cladmi) [reply makefiles/arch/cortexm.inc.mk]: I tried to be consistent with the other handling of WERROR

https://github.com/RIOT-OS/RIOT/blob/6a8808c5113d6ca7f5f62293a64db89393972aa1/Makefile.include#L141-L143

**Comment by cladmi** [comment]
On my machine I get the message… I will try with the vagrant image.

What do you get with 

```RIOT_CI_BUILD=1 BOARD=samr21-xpro WERROR=0 make --no-print-directory -C examples/default/ info-debug-variable-ARM_GCC_VERSION` ?

**Comment by cladmi** [comment]
It correctly fails with the test command in a vagrant image. Did you remember to checkout the branch before testing in the VM ?
I did same mistake on first test…

File: makefiles/arch/cortexm.inc.mk (hunk 2 above)
Comment (miri64) [reply makefiles/arch/cortexm.inc.mk]: Mhhhh... Ok for now, but we should think about if it isn't more clearer the other way around (afaics only `WERROR=0` is documented, nothing is said about `WERROR=<random number != 0>`

**Comment by miri64** [comment]
> It correctly fails with the test command in a vagrant image. Did you remember to checkout the branch before testing in the VM ?
> I did the mistake on first test…

Pretty sure I did, but let me try again ;-)

**Comment by miri64** [comment]
> > It correctly fails with the test command in a vagrant image. Did you remember to checkout the branch before testing in the VM ?
> > I did the mistake on first test…
> 
> Pretty sure I did, but let me try again ;-)

Arghs, though I mounted the repo from the host machine, I hooked it into a different repo then the one I checked out the PR in (I copied the config for the VM from my #9942 test setup where I did not want to interfere with my development cycle).

**Comment by miri64** [comment]
Now it works as expected. Sorry for the noise -.-

**Comment by miri64** [comment]
(please squash)

**Comment by miri64** [comment]
Then let's go

**Comment by cladmi** [comment]
Thank you for reviewing and testing.
